src/main/java/io/openliberty/tools/common/plugins/util/BinaryScannerUtil.java

```diff
@@ -0,0 +1,263 @@
+                        throw new RecommendationSetException(true, conflicts, sampleFeatureList);
+                    }
+                } else if (scannerException.getClass().getName().endsWith("FeatureConflictException")) {
+                    // The scanned files conflict with each other or with current features
+                    Set<String> conflicts = getConflicts(scannerException);
+                    Set<String> sampleFeatureList = null;
+                    if (currentFeatureSet != null) {
+                        try {
+                            sampleFeatureList = runBinaryScanner(null, classFiles, allClassesDirectories, eeVersion, mpVersion, true);
+                        } catch (InvocationTargetException retryException) {
+                            Throwable scannerSecondException = retryException.getCause();
+                            if (scannerSecondException.getClass().getName().endsWith("FeatureConflictException")) {
+                                // Even after removing the server.xml feature list there are still conflicts in the binaries
+                                throw new NoRecommendationException(conflicts);
+                            } else {
+                                debug("Unexpected failure on retry call to binary scanner", scannerSecondException);
+                                debug("Passed directories to binary scanner:"+allClassesDirectories);
+                                sampleFeatureList = getNoSampleFeatureList();
+                            }
+                        }
+                        throw new RecommendationSetException(false, conflicts, sampleFeatureList);
+                    } else {
+                        throw ite;
+                    }
+                } else {
+                    //TODO handle more exceptions from binary scanner e.g. com.ibm.ws.report.exceptions.RequiredFeatureModifiedException
+                    error(scannerException.getMessage());
+                    debug("Exception from binary scanner.", scannerException);
+                }
+            } catch (MalformedURLException|ClassNotFoundException|NoSuchMethodException|IllegalAccessException x){
+                // TODO Figure out what to do when there is a problem scanning the features
+                error("Exception:"+x.getClass().getName());
+                Object o = x.getCause();
+                if (o != null) {
+                    warn("Caused by exception:"+x.getCause().getClass().getName());
+                    warn("Caused by exception message:"+x.getCause().getMessage());
+                }
+                error(x.getMessage());
+            }
+        } else {
+            if (binaryScanner == null) {
+                throw new PluginExecutionException("The binary scanner jar location is not defined.");
+            } else {
+                throw new PluginExecutionException("Could not find the binary scanner jar at " + binaryScanner.getAbsolutePath());
```

Comment:
As we discussed on the call, I don't think we should throw an exception when the binary scanner is not found. Instead log a warning that the features could not be generated because the binary scanner could not be located.

Reply:
This will be addressed as part of this issue: https://github.com/OpenLiberty/ci.maven/issues/1291